jest/index.d.ts

```diff
@@ -147,11 +188,28 @@ declare namespace jest {
         new (...args: any[]): any
     }
 
-    interface Mock<T> extends Function {
+    interface Mock<T> extends Function, MockInstance<T> {
         new (): T;
         (...args: any[]): any;
+    }
+
+    /**
+     * Wrap module with mock definitions
+     * @example
+     *  jest.mock("../api");
+     *  import { Api } from "../api";
+     *
+     *  const myApi: jest.Mocked<Api> = new Api() as any;
+     *  myApi.myApiMethod.mockImplementation(() => "test");
+     */
+    type Mocked<T> = T & {
+        [P in keyof T]: T[P] & MockInstance<T>;
```

Comment:
No. MockInstance only adds few mock methods. T[P] is needed to keep original method signature